### PR TITLE
child workflow execution parent close policy

### PR DESCRIPTION
**Pull request by aryak-stripe**
`Temporal::Workflow.execute_workflow` spawns a workflow, but doesn't currently include setting the ParentClosePolicy, instead defaulting to TERMINATE. This PR adds support for `parent_close_policy` in this call, along with integration specs of this behavior.

### Test Plan:
```
cd examples
bundle exec rspec spec/integration/parent_close_workflow_spec.rb
bundle exec rspec spec/
cd ..
bundle exec rspec spec/unit
```

### CODE REVIEW

**Comment by DeRauk** [comment]
Adding @antstorm as a reviewer too.

**Review by antstorm — APPROVED** [review]
Nice one, thank you @aryak-stripe 👍 One small thing that needs to be removed, but otherwise ready to merge

File: examples/spec/integration/parent_close_workflow_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+require 'workflows/parent_close_workflow'
+
+describe ParentCloseWorkflow, :integration do
+  subject { described_class }
+
+  before { allow(SlowChildWorkflow).to receive(:stub).and_call_original }
```
[the remark this answers is not in the document]
Comment (aryak-stripe) [reply examples/spec/integration/parent_close_workflow_spec.rb:6]: Oops! You're right, I'll adjust that now

**Review by antstorm — APPROVED** [review]
🙌

**Pull request merged**